pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -225,6 +232,97 @@ def set_config(self, cfg: SimConfig) -> None:
                 for qid in self._qid_index
             ]
 
+        if "depolarizing" in self.config.noise:
+            if self.basis_name == "digital" or self.basis_name == "all":
+                # Go back to previous config
+                self.set_config(prev_config)
+                raise NotImplementedError(
+                    "Cannot include depolarizing "
+                    + "noise in digital- or all-basis."
+                )
+
+            # Probability of error occurrence
+
+            prob = self.config.depolarizing_prob
+            n = self._size
+            if prob > 0.1 and n > 1:
+                warnings.warn(
+                    "The depolarizing model is a first-order approximation"
+                    f" in the depolarizing probability. p = {2*prob}"
+                    " is too large for realistic results.",
+                    stacklevel=2,
+                )
+
+            k = np.sqrt((prob / 3) * (1 - prob) ** (n - 1))
+            self._collapse_ops = [
+                np.sqrt((1 - prob) ** n)
+                * qutip.tensor([self.op_matrix["I"] for _ in range(n)])
+            ]
+            self._collapse_ops += [
+                k
+                * (
+                    self.build_operator([("sigma_rr", [qid])])
+                    - self.build_operator([("sigma_gg", [qid])])
+                )
+                for qid in self._qid_index
+            ]
+            self._collapse_ops += [
+                k
+                * (
+                    self.build_operator([("sigma_gr", [qid])]).dag()
+                    + self.build_operator([("sigma_gr", [qid])])
+                )
+                for qid in self._qid_index
+            ]
+            self._collapse_ops += [
+                1j
+                * k
+                * (
+                    self.build_operator([("sigma_gr", [qid])]).dag()
+                    - self.build_operator([("sigma_gr", [qid])])
+                )
+                for qid in self._qid_index
+            ]
```

Comment:
Where did you get these Kraus operators from? They don't seem right to me

Reply:
Or rather, they make sense if you make `p -> p*3/4`, I think you are missing that

Reply:
![image](https://user-images.githubusercontent.com/76184755/211568312-c383a0b2-2198-4e89-b6b6-13c2a086f55e.png)

Reply:
From chapter 8 on a MIT lecture on radiation.
Link: https://ocw.mit.edu/courses/22-51-quantum-theory-of-radiation-interactions-fall-2012/resources/mit22_51f12_ch8/

Reply:
I tried then to add, M1, M2, and M3 according the uncorrelated noise approximation for many-body systems, i.e. assuming one bit is modified at the time.

Reply:
Ok, it seems that we have contradicting sources. Yours matches what is in Nielsen and Chuang, so I think we should stick to it

Reply:
But wait, wikipedia says otherwise... https://en.wikipedia.org/wiki/Quantum_depolarizing_channel

Reply:
Ok, I think the issue is the definition. My sources define "depolarization" as _"an operation that replaces the input state with a completely mixedstate with probability p"_, while yours state that _"with probability p an “error” occurs.
The error can be of any one of three types, where each type of error is equally likely."_ 

These are different things, so it's natural that they have different formulas.

Reply:
I browse the Nielsen and Chuang book on Quantum COmputation and Quantum Information and they have a slight different set of operators.

Reply:
And look what they say right after that: 
<img width="676" alt="image" src="https://user-images.githubusercontent.com/29920212/211784578-bbe6879b-4a7b-4002-bca8-851a4f93dc9a.png">

This is essentially our problem here. I think we just have to pick one and clearly state what we mean by "depolarizing" in the docstring.

Reply:
Defining the "depolarization" as "an operation that replaces the input state with a completely mixed state with probability p" is not wrong. We can start from that definition and end up with the Krauss operators I sent first.

Reply:
Let's see what @sebgrijalva thinks

Reply:
@sebgrijalva told me he prefers the equation (8.102) which is in accordance with the dephasing model already chosen.